internal/literals/literals.go

```diff
@@ -130,37 +130,42 @@ func Obfuscate(file *ast.File, info *types.Info, fset *token.FileSet, linkString
 		return true
 	}
 
-	// Imports which are used might be marked as unused by only looking at the ast,
-	// because packages can declare a name different from the last element of their import path.
-	//
-	// 	package main
-	//
-	// 	import "github.com/user/somepackage"
-	//
-	//	func main(){
-	//		// this line uses github.com/user/somepackage
-	//		anotherpackage.Foo()
-	//	}
-	//
-	// TODO: remove this check and detect used imports with go/types somehow
-	prevUsedImports := make(map[string]bool)
-	for _, imp := range file.Imports {
-		path, err := strconv.Unquote(imp.Path.Value)
-		if err != nil {
-			panic(err)
-		}
-		prevUsedImports[path] = astutil.UsesImport(file, path)
-	}
-
 	file = astutil.Apply(file, pre, post).(*ast.File)
 
 	// some imported constants might not be needed anymore, remove unnecessary imports
+	usedImports := make(map[string]bool)
+	ast.Inspect(file, func(n ast.Node) bool {
+		node, ok := n.(*ast.SelectorExpr)
+		if !ok {
+			return true
+		}
+
```

Comment:
I believe you don't need this at all; `ast.Inspect` will walk the entire syntax tree, so you'll already walk all identifiers. you just need a `func(node ast.Node)` that filters by `*ast.Ident` and then uses `info.Uses`.

Reply:
Right